### ISSUE TITLE
False positive for html-has-lang and 'meta-viewport'

### ISSUE THREAD

**Issue by pjhumkhawala** [dequelabs/axe-core#1227]
I am developing an app for my company on Angular 7 and while performing unit tests on the components, axe core throws the following error. However, I already have the `lang` attribute in the <html> tag and the meta tag is also not setting `user-scalable=no`. When I use the axe extension on the chrome browser, it does not throw that error. I was not sure if this is a bug or it is just that my `<html>` and `<header>` tags are in a separate file and I am testing just the component.

Error thrown:
```
       Failed: [
            {
                "id": "html-has-lang",
                "impact": "serious",
                "tags": [
                    "cat.language",
                    "wcag2a",
                    "wcag311"
                ],
                "description": "Ensures every HTML document has a lang attribute",
                "help": "<html> element must have a lang attribute",
                "helpUrl": "https://dequeuniversity.com/rules/axe/3.1/html-has-lang?application=axeAPI",
                "nodes": [
                    {
                        "any": [
                            {
                                "id": "has-lang",
                                "data": null,
                                "relatedNodes": [],
                                "impact": "serious",
                                "message": "The <html> element does not have a lang attribute"
                            }
                        ],
                        "all": [],
                        "none": [],
                        "impact": "serious",
                        "html": "<html>",
                        "target": [
                            "html"
                        ],
                        "failureSummary": "Fix any of the following:\n  The <html> element does not have a lang attribute"
                    }
                ]
            },
            {
                "id": "meta-viewport",
                "impact": "critical",
                "tags": [
                    "cat.sensory-and-visual-cues",
                    "wcag2aa",
                    "wcag144"
                ],
                "description": "Ensures <meta name=\"viewport\"> does not disable text scaling and zooming",
                "help": "Zooming and scaling must not be disabled",
                "helpUrl": "https://dequeuniversity.com/rules/axe/3.1/meta-viewport?application=axeAPI",
                "nodes": [
                    {
                        "any": [
                            {
                                "id": "meta-viewport",
                                "data": "user-scalable=no",
                                "relatedNodes": [],
                                "impact": "critical",
                                "message": "user-scalable=no on <meta> tag disables zooming on mobile devices"
                            }
                        ],
                        "all": [],
                        "none": [],
                        "impact": "critical",
                        "html": "<meta name=\"viewport\" content=\"width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no\">",
                        "target": [
                            "meta[name=\"viewport\"]"
                        ],
                        "failureSummary": "Fix any of the following:\n  user-scalable=no on <meta> tag disables zooming on mobile devices"
                    }
                ]
            }
        ]
```

HTML content (minimal code which I can send)

```
<html lang="en">

<head>
  <script>
    __Zone_enable_cross_context_check = true;
    </script>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
```
<pre><code>
axe-core version: 3.1.2
axe-webdriver, extension or other integration version: 3.6.0

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 10  
- Platform:  Windows 
</code></pre>

**Comment by EzraBrooks** [comment]
Karma and Jasmine insert the Angular component directly into their own HTML document, which is where you're getting the warning from. In the scope of unit tests, you can probably safely disable these particular warnings.

**Comment by pjhumkhawala** [comment]
Yes! That is exactly what I ended up doing!

**Comment by marcysutton** [comment]
@pjhumkhawala are you providing a context for your unit tests that limits the scope to individual components? Because it sounds like you're running everything on a component including page-level rules, when those won't apply. I'd recommend checking out the API documentation on context for `axe.run`. https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#api-name-axerun
